From 537b3c58423328a3e1121ea3d9c93b49df3d9430 Mon Sep 17 00:00:00 2001
From: Ley Foon Tan <ley.foon.tan@intel.com>
Date: Fri, 26 Feb 2021 11:40:20 +0800
Subject: [PATCH 114/120] HSD #1508890903: net: ethernet: altera: Fixes size of
 DMA buffer allocation

commit ca8a9592aa367a5db7a3c23f6c7afcd6dd4cc4a3 from
https://github.com/altera-opensource/linux-socfpga.git

Fixes 'struct altera_dma_private' to 'struct altera_dma_buffer'.

Allocates the buffer based on size of struct altera_dma_buffer instead
of struct altera_dma_private.

Signed-off-by: Patrick Heffernan <pheffernan@t3innovation.com>
Signed-off-by: Ley Foon Tan <ley.foon.tan@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c b/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
index 0bd1d18230c8..0b5fbc67800a 100644
--- a/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
+++ b/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
@@ -162,14 +162,14 @@ int alloc_init_skbufs(struct intel_fpga_qse_private *priv)
 
 	/* Create Rx ring buffer */
 	priv->dma_priv.rx_ring = kcalloc(rx_descs,
-					 sizeof(struct altera_dma_private),
+					 sizeof(struct altera_dma_buffer),
 					 GFP_KERNEL);
 	if (!priv->dma_priv.rx_ring)
 		goto err_rx_ring;
 
 	/* Create Tx ring buffer */
 	priv->dma_priv.tx_ring = kcalloc(tx_descs,
-					 sizeof(struct altera_dma_private),
+					 sizeof(struct altera_dma_buffer),
 					 GFP_KERNEL);
 	if (!priv->dma_priv.tx_ring)
 		goto err_tx_ring;
-- 
2.31.1

